### ISSUE TITLE
NonEmptyList extension functions

### ISSUE THREAD

**Issue by pablisco** [arrow-kt/arrow#1304]
Add extension functions for factory methods so we can do:
```
list.toNel()
```
Instead of:
```
NonEmptyList.fromList(list)
```

**Comment by raulraja** [comment]
Isn't that an unsafe operation? what happens if the List is empty?

**Comment by fedefernandez** [comment]
I guess it should return an `Option<NonEmptyList>` like in cats

**Comment by nomisRev** [comment]
@raulraja there should be `fromList` and `unsafeFromList` although possible that the second one is internal.

**Comment by pablisco** [comment]
Yeah, both are available so I was thinking that we could have something like this:
```
inline fun <A> List<A>.toNel() : Option<NonEmptyList<A>>  = 
  NonEmptyList.fromList(this)

inline fun <A> List<A>.toUnsafeNel() : NonEmptyList<A> = 
  NonEmptyList.unsafeFromList(this)
```

**Comment by pakoito** [comment]
We don't have unsafe functions on the libraries. I believe we considered an arrow-unsafe module a while ago, but yeah, better stick to the safe versions for now.
